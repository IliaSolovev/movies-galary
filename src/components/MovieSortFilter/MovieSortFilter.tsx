import React from 'react';

import FilterButton from './FilterButton/FilterButton';
import { Filters } from '../../redux/moviesSlice';

import style from './movieSortFilter.module.scss';

interface Props {
    movieCount: number,
    onSetMoviesSortFilter: (filter: Filters) => void,
    currentFilter: Filters
}

export const MovieSortFilter: React.FC<Props> = ({ movieCount, onSetMoviesSortFilter, currentFilter }) => {
  const filterButtons = ['release date', 'rating'].map((filter: Filters, id) => (
    <FilterButton
      text={filter}
      key={id}
      isActive={currentFilter === filter}
      onClick={() => onSetMoviesSortFilter(filter)}
    />
  ));
  return (
    <div className={style.movieSortFilter}>
      <div>
        <div className={style.movieSortFilter__counter}>
          {movieCount}
          {' '}
          movies found
        </div>
        <div className={style.movieSortFilter__filters}>
          <div>Sort by</div>
          <div className={style.movieSortFilter__filterButton}>
            {filterButtons}
          </div>
        </div>
      </div>
    </div>
  );
};
