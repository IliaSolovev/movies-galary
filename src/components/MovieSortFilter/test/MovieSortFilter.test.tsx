import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { MovieSortFilter } from '../MovieSortFilter';
import { Button } from '../..';

it('should render sort filter', () => {
  const component = shallow(<MovieSortFilter currentFilter="rating" movieCount={6} onSetMoviesSortFilter={() => {}} />);
  expect(toJson(component)).toMatchSnapshot();
});
it('should render sort button', () => {
  const component = shallow(<Button type="sortFilter" active onClick={() => {}}>sort</Button>);
  expect(toJson(component)).toMatchSnapshot();
});
