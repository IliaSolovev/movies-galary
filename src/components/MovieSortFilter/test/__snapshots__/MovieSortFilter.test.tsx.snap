// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render sort button 1`] = `
<button
  className="button button_active"
  onClick={[Function]}
>
  asd
</button>
`;

exports[`should render sort filter 1`] = `
<div
  className="movieSortFilter"
>
  <div>
    <div
      className="movieSortFilter__counter"
    >
      6
       
      movies found
    </div>
    <div
      className="movieSortFilter__filters"
    >
      <div>
        Sort by
      </div>
      <div
        className="movieSortFilter__filterButton"
      >
        <Component
          active={false}
          key="0"
          onClick={[Function]}
          type="sortFilter"
        >
          release date
        </Component>
        <Component
          active={true}
          key="1"
          onClick={[Function]}
          type="sortFilter"
        >
          rating
        </Component>
      </div>
    </div>
  </div>
</div>
`;
